go-controller/pkg/crd/userdefinednetwork/v1/cudn.go

```diff
@@ -26,6 +26,7 @@ type ClusterUserDefinedNetworkSpec struct {
 	// NamespaceSelector Label selector for which namespace network should be available for.
 	// +kubebuilder:validation:Required
 	// +required
+	// +kubebuilder:validation:XValidation:rule="has(self.matchLabels) && self.matchLabels.size() > 0 || has(self.matchExpressions) && self.matchExpressions.size() > 0", message="namespace selector must have at least one rule"
```

Comment:
coming here with fresh mind but looking at the message "namespace selector must have at least one rule" means you have to select something for sure..

but it doesn't reflect the problem which is if I do have 1 rule and that rule is MatchLabels: {} then I think what you are trying to fix is "selecting all namespaces is not allowed like in your tests right?"

for user facing side is that more clearer a message than what you have here "have more than 1 rule?"